atomic_reactor/util.py

```diff
@@ -320,14 +323,27 @@ def clone_git_repo(git_url, target_dir, commit=None, retry_times=GIT_MAX_RETRIES
     logger.debug("url = '%s', dir = '%s', commit = '%s'",
                  git_url, target_dir, commit)
 
-    cmd = ["git", "clone", git_url, quote(target_dir)]
+    cmd = ["git", "clone"]
+    if branch:
+        cmd += ["-b", branch, "--single-branch"]
+    if depth:
+        cmd += ["--depth", '{0}'.format(depth)]
+    cmd += [git_url, quote(target_dir)]
 
     logger.debug("cloning '%s'", cmd)
+    repo_commit = ''
     for counter in range(retry_times + 1):
         try:
             # we are using check_output, even though we aren't using
             # the return value, but we will get 'output' in exception
             subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+            repo_commit = reset_git_repo(target_dir, commit)
+            if depth and repo_commit != commit:
```

Comment:
I think if the commit didn't get fetched we'll get an exception from `reset_git_repo` won't we?
```
$ git reset --hard f00000
fatal: ambiguous argument 'f00000': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
[twaugh@river atomic-reactor]$ echo $?
128
```

Reply:
Fixed.